adapters/Vungle/VungleAdapter/GADMAdapterVungleRouter.m

```diff
@@ -519,15 +550,19 @@ - (void)vungleAdPlayabilityUpdate:(BOOL)isAdPlayable
 
   if (isAdPlayable) {
     [delegate adAvailable];
-  } else if (!isAdPlayable && ![[VungleSDK sharedSDK] isAdCachedForPlacementID:placementID]) {
-    if (error) {
-      NSLog(@"Vungle Ad Playability returned an error: %@", error.localizedDescription);
-    } else {
-      error = GADMAdapterVungleErrorWithCodeAndDescription(kGADErrorMediationAdapterError, [NSString stringWithFormat:@"Ad is not available for placementID: %@.", placementID]);
-    }
-    [delegate adNotAvailable:error];
-    [self removeDelegate:delegate];
+    return;
   }
+
+  // Ad not playable. Return an error.
```

Comment:
@ericleich Hi Eric, I think we need to check `[[VungleSDK sharedSDK] isAdCachedForPlacementID:placementID]` here like previous implementation, because when we call `- (BOOL)playAd:options:placementID:error:` method of VungleSDK to play a fullscreen ad , before playing the ad, VungleSDK will call this `- (void)vungleAdPlayabilityUpdate:placementID:error:` delegate method and pass `NO` for the `isAdPlayable` parameter.  In this case, we should not call `[delegate adNotAvailable:error];` and `[self removeDelegate:delegate];` methods, which will cause many issues. Thanks a lot and have a great day！^_^

Reply:
@YueVungle Thanks for the feedback! I see now there is a disconnect between the router getting playabilityUpdate, and trying to tell adapters whether ad is available or not. But VungleRouter doesn't know about presentation state.

I plan to solve this by still forwarding adNotAvailable to adapters, but having the adapters check if the ad is presenting or not, and ignore the error if it's already presenting. Note: We already do the same for adAvailable in GADMAdapterVungleInterstitial and GADMAdapterVungleRewardedAd.

I'll make updates and share with you soon.

Reply:
Also FYI, GMA SDK would ignore a load error anyways if the adapter is already presenting. But it's better for that to be explicit in the adapter.

Reply:
@YueVungle code updated, please take another look.

I redefined the "presenting" status in GADMAdapterVungleInterstitial and GADMAdapterVungleRewardedAd to just be an "ad loaded" status, because we are using it to avoid sending ad load success/fail callbacks to adapters that have already loaded. This also addresses banner cases as well.

Reply:
@ericleich Hi Eric, thanks for the update, but as I mentioned above, when we call `- (BOOL)playAd:options:placementID:error: method` of VungleSDK to play a fullscreen ad , before playing the ad, VungleSDK will call this `- (void)vungleAdPlayabilityUpdate:placementID:error:` delegate method and pass `NO` for the `isAdPlayable` parameter.  In this case, we cannot call `[self removeDelegate:delegate];`，this will prevent the delegate from receiving many important callbacks.  Have a nice Friday! O(∩_∩)O

Reply:
@YueVungle Thanks for the feedback. I added back in a check to see if the ad is cached. Please take another look.

Note: I am confused as to why the isAdCachedForPlacementID:placementID check returns YES, when isAdPlayable returns NO. The docs for that method say `Returns YES  when there is certainty that an ad will be able to play for a given placementID.`, but I'm getting a callback saying isAdPlayable=NO...

However, that does seem to be the behavior, and it would be complex to keep track of even more state in the adapter, so I stuck with that implementation.

Reply:
@ericleich Hi Eric, thanks so much for adding the check for ad is cached back. I'm really sorry to make you confused, we will optimize our SDK in future release to make a better designed API. I saw your comments for checking ad is cached,  which is very detailed and clear. 
Now the PR looks good to me, thanks a lot again for all the help and support in our Adapter implementation and issue investigation. I really appreciate it. Have a great day! 😊

Reply:
Thanks for the review @YueVungle and for catching the above errors!